redash/tasks/worker.py

```diff
@@ -22,10 +23,50 @@ def is_cancelled(self):
         return self.meta.get("cancelled", False)
 
 
+class StatsdRecordingQueue(BaseQueue):
+    """
+    RQ Queue Mixin that overrides `enqueue_call` to increment metrics via Statsd
+    """
+
+    def enqueue_call(self, func, args=None, kwargs=None, timeout=None,
+                     result_ttl=None, ttl=None, failure_ttl=None,
+                     description=None, depends_on=None, job_id=None,
+                     at_front=False, meta=None):
+        job = super().enqueue_call(
+            func, args, kwargs, timeout,
+            result_ttl, ttl, failure_ttl, description,
+            depends_on, job_id, at_front, meta
+        )
+        statsd_client.incr("rq.jobs.created.{}".format(self.name))
+        return job
+
+
 class CancellableQueue(BaseQueue):
     job_class = CancellableJob
 
 
+class RedashQueue(StatsdRecordingQueue, CancellableQueue):
+    pass
+
+
+class StatsdRecordingWorker(BaseWorker):
+    """
+    RQ Worker Mixin that overrides `execute_job` to increment/modify metrics via Statsd
+    """
+
+    def execute_job(self, job, queue):
+        statsd_client.incr("rq.jobs.running.{}".format(queue.name))
+        statsd_client.incr("rq.jobs.started.{}".format(queue.name))
+        try:
+            super().execute_job(job, queue)
+        finally:
+            statsd_client.decr("rq.jobs.running.{}".format(queue.name))
+            if job.get_status() == JobStatus.FINISHED:
+                statsd_client.incr("rq.jobs.finished.{}".format(queue.name))
+            else:
+                statsd_client.incr("rq.jobs.failed.{}".format(queue.name))
```

Comment:
There are other statuses in rq that this should be tracking, so I would suggest to make the condition:

 ```suggestion
            if job.get_status() == JobStatus.FAILED:
                statsd_client.incr("rq.jobs.failed.{}".format(queue.name))
            elif job.get_status() == JobStatus.DEFERRED:
                statsd_client.incr("rq.jobs.deferred.{}".format(queue.name))
            # TODO: add more status conditions here
            # elif ...
            else:
                statsd_client.incr("rq.jobs.finished.{}".format(queue.name))
```

BUT we

Reply:
Are those other statuses terminal? I was intending to only track terminal statuses in the metrics, and classify anything left in statuses like `STARTED` or `QUEUED` to be a "failed" job, since by my eye that's how the `monitor_work_horse` function seems to treat those jobs.

Reply:
So DEFERRED means if a job has a dependency then it's not being processed if the job being depended upon has not been finished yet. So it's not "failed" but just unprocessed yet. @arikfr I would wait for this to be fixed before merging.